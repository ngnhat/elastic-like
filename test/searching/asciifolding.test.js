/**
 * Created by ngnhat on Sat May 25 2019
 */
const Store = require('../../index');

describe('vietnamese:', () => {
  it('buổi trưa ăn bưởi chua: standard', () => {
    const store = new Store();
    store.add({ Id: 1, Code: 'code001', Name: 'buổi trưa ăn bưởi chua' });

    expect(store.search('bưởi')).toMatchSnapshot();
    expect(store.search('buổi')).toMatchSnapshot();
    expect(store.search('buoi')).toMatchSnapshot();
  });

  it('buổi trưa ăn bưởi chua: asciifolding', () => {
    const store = new Store({
      mapping: {
        Code: { type: 'text', analyzer: 'standard' },
        Name: { type: 'text', analyzer: 'asciifolding' },
      },
    });
    store.add({ Id: 1, Code: 'code001', Name: 'buổi trưa ăn bưởi chua' });

    expect(store.search('bưởi')).toMatchSnapshot();
    expect(store.search('buổi')).toMatchSnapshot();
    expect(store.search('buoi')).toMatchSnapshot();
  });
});
