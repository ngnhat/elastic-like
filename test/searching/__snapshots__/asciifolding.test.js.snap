// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vietnamese: buổi trưa ăn bưởi chua: asciifolding 1`] = `
Array [
  Object {
    "score": 0.39556284962119864,
    "source": Object {
      "Code": "code001",
      "Id": 1,
      "Name": "buổi trưa ăn bưởi chua",
    },
  },
]
`;

exports[`vietnamese: buổi trưa ăn bưởi chua: asciifolding 2`] = `
Array [
  Object {
    "score": 0.39556284962119864,
    "source": Object {
      "Code": "code001",
      "Id": 1,
      "Name": "buổi trưa ăn bưởi chua",
    },
  },
]
`;

exports[`vietnamese: buổi trưa ăn bưởi chua: asciifolding 3`] = `
Array [
  Object {
    "score": 0.39556284962119864,
    "source": Object {
      "Code": "code001",
      "Id": 1,
      "Name": "buổi trưa ăn bưởi chua",
    },
  },
]
`;

exports[`vietnamese: buổi trưa ăn bưởi chua: standard 1`] = `
Array [
  Object {
    "score": 0.28768207245178085,
    "source": Object {
      "Code": "code001",
      "Id": 1,
      "Name": "buổi trưa ăn bưởi chua",
    },
  },
]
`;

exports[`vietnamese: buổi trưa ăn bưởi chua: standard 2`] = `
Array [
  Object {
    "score": 0.28768207245178085,
    "source": Object {
      "Code": "code001",
      "Id": 1,
      "Name": "buổi trưa ăn bưởi chua",
    },
  },
]
`;

exports[`vietnamese: buổi trưa ăn bưởi chua: standard 3`] = `Array []`;
